pkg/translator/prometheusremotewrite/metrics_to_prw_v2.go

```diff
@@ -131,9 +141,40 @@ func (c *prometheusConverterV2) addSample(sample *writev2.Sample, lbls []prompb.
 		off = c.symbolTable.Symbolize(l.Value)
 		buf = append(buf, off)
 	}
-	ts := writev2.TimeSeries{
+
+	sig := timeSeriesSignature(lbls)
+	ts := &writev2.TimeSeries{
 		LabelsRefs: buf,
 		Samples:    []writev2.Sample{*sample},
 	}
-	c.unique[timeSeriesSignature(lbls)] = &ts
+
+	// check if the time series is already in the unique map
+	if existingTS, ok := c.unique[sig]; ok {
+		// if the time series is already in the unique map, check if it is the same metric
+		if !isSameMetricV2(existingTS, ts) {
+			// if the time series is not the same metric, add it to the conflicts map
```

Comment:
This comment also replicates what the code does, it will just make maintenance more difficult.
```suggestion
```